rfd/9999-aws-api-integration-using-oidc.md

```diff
@@ -0,0 +1,554 @@
+  - `public`: we can either return clear text version of the subject (user names) or
+  - `pairwise`: we can return a hash of the subject's identity (to hide their email, for instance)
+
+##### JSON Web Key Sets
+This endpoint returns a list of public keys (usually only one key, except for periods of key rotation).
+
+This endpoint URI must be equal to `jwks_uri` defined in the previous section.
+
+As an example, `https://proxy.example.com/.well-known/jwks-oidc`.
+
+It should return the following:
+
+```
+200 OK
+
+{
+  "keys": [
+    {
+      "kty": "RSA",
+      "alg": "RS256",
+      "n": "<public key value>",
+      "e": "AQAB"
+    }
+  ]
+}
+```
+
+This must respect the RFC7517 (eg of a key can be found at https://www.rfc-editor.org/rfc/rfc7517#appendix-A.1).
+
+Obtaining this key should be possible using
+```golang
+GetCertAuthority(
+	ctx,
+	types.CertAuthID{
+		Type:       types.OIDCSigner, // new key type
+		DomainName: clusterName,
+	},
+	false /*loadKeys*/,
+)
+```
+
+#### AWS Role for Teleport OIDC IdP
+
+The user will create or associate an AWS Role to the Teleport OIDC IdP.
+
+As an example, we the following is the required policy to list RDS DBs (including Aurora clusters):
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Actions": "[
+				rds:DescribeDBClusters",
+				rds:DescribeDBInstances",
+	        ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+This AWS Role must trust Teleport as an IdP.
+To do so, the user must add the following Trusted relationship:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::0123456789012:oidc-provider/proxy.example.com"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "proxy.example.com:aud": "sts.amazonaws.com"
+                }
+            }
+        }
+    ]
+}
+```
+
+### UX
+Besides the flow describe above (Users Point Of View), the Integration resource will have the usual CRUD operations via both API and `tctl`.
+The only updatable field is `aws_role_arn`.
+
+HTTP API:
+```
+Methods:
+GET .../integration
+GET .../integration/:id
+POST .../integration
+PUT .../integration/:id
+DELETE .../integration/:id
+```
+JSON representation:
+```json
+{
+	"name": "myaws",
+	"subkind": "aws-oidc"
+	"subkindData": {
+		"aws_role_arn": "arn:aws:123:TeleportOIDC"
+	}
+}
+```
+
+`tctl`:
+
+```
+$ tctl get integrations
+kind: integration
+subkind: aws-oidc
+version: v1
+metadata:
+	name: some-name
+spec:
+	aws_role_arn: arn:aws:123:TeleportOIDC
+
+$ tctl get integration/myaws
+kind: integration
+subkind: aws-oidc
+version: v1
+metadata:
+	name: some-name
+spec:
+	aws_role_arn: arn:aws:123:TeleportOIDC
+
+$ tctl create aws-integration.yaml
+```
+
+Resource representation as `yaml`:
```

Comment:
Can you add a section covering IaC flows to the RFD as well? Are there any changes needed to Terraform provider, Kube operator and Helm charts?

Reply:
Yes :+1: